pkg/alibabacloud/api/api.go

```diff
@@ -396,32 +396,29 @@ func (c *Client) UnassignPrivateIPAddresses(ctx context.Context, eniID string, a
 	return err
 }
 
-func (c *Client) describeNetworkInterfaces(ctx context.Context, subnets ipamTypes.SubnetMap) ([]ecs.NetworkInterfaceSet, error) {
+func (c *Client) describeNetworkInterfaces(ctx context.Context) ([]ecs.NetworkInterfaceSet, error) {
 	var result []ecs.NetworkInterfaceSet
 
-	for _, subnet := range subnets {
-		for i := 1; ; {
-			req := ecs.CreateDescribeNetworkInterfacesRequest()
-			req.PageNumber = requests.NewInteger(i)
-			req.PageSize = requests.NewInteger(50)
-			req.VSwitchId = subnet.ID
```

Comment:
It's not clear from context: Why is this unnecessary? Won't this increase the number of returned instances?

Reply:
Oh I see, that’s true. Ideally we should be able to filter out cilium relevant resources in some way. But looks like for now we are listing all resources arbitrarily:
https://github.com/cilium/cilium/blob/ecea62eea4b4aba593747f22f7371f377bf247eb/pkg/alibabacloud/eni/instances.go#L90
and in this context, iterating over subnets doesn’t help but introduces more requests, if there is many subnets.

Reply:
So on Amazon ENI, we have the instance-tag-filter since Cilium v1.12. This might be something that could be used here.

The subnets passed into the API call before were also coming from the subnet-tag-filter, which is something users can explicitly set to filter out irrelevant resources. Therefore I would consider keeping it in.

Reply:
Agreed with the instance-tag-filter bits.

>The subnets passed into the API call before were also coming from the subnet-tag-filter, which is something users can explicitly set to filter out irrelevant resources.

Looks like this is only the case for AWS, but for alibabacloud, we are listing all subnets:
https://github.com/cilium/cilium/blob/88f04b69ea2f96d3c16cf7b1798bc178bb50d4b4/pkg/alibabacloud/api/api.go#L94-L95  
so I think the removal of the subnet filter should not harm, but if we are nervous about removing it, I guess we could drop the commit from this PR and address the resource filtering in a separate PR. What do you think?